### ISSUE TITLE
Multi-root workspace and icon

### ISSUE THREAD

**Issue by zabel-xyz** [Microsoft/vscode#38145]
- VSCode Version: 1.18.1
- OS Version: Windows 10

I use muli-root workspace, and lost the specific icon defined for some folders:

![image](https://user-images.githubusercontent.com/16486407/32697388-caffcc5a-c78f-11e7-8b12-f2fa956e1e9f.png)

I use my own icon theme and specify
```
    "folderNames": {
        "db": "_fd_red",
        "server": "_fd_green"
    }
```
Now, with multi-workspace, they are no longer used.

There is a setting `rootFolder` but no` rootFolderNames`. 

I would like a setting taking account **name of folder** or **name defined in .code-workspace.** to get something like this:
![image](https://user-images.githubusercontent.com/16486407/32697563-54be0c38-c793-11e7-8f08-b029157b792c.png)

**Comment by aeschli** [comment]
'rootFolderNames' is currently not planned, unless there are more requests for it.

**Comment by amitzur** [comment]
I want this feature as well. 
Are you in favor of this API for icon themes? Would a PR to support `rootFolderNames` help move this forward?

**Comment by aeschli** [comment]
Yes. PR is welcome. I suggest `workspaceFolder`. No sure if name specific icons are really important here, but if you think so it would be `workspaceFolderNames`.

**Comment by amitzur** [comment]
so change the current `rootFolder` setting? It will be a breaking change. Although it's undocumented AFAICT

**Comment by aeschli** [comment]
Oh, you are right, forgot about that one. `rootFolderNames` then.
Also need to document 'rootFolder'

**Comment by aeschli** [comment]
it's been implemented! As of VS Code 1.85 (commit 2b9791e491a, polished in 272653c995c), icon themes can use rootFolderNames and rootFolderNamesExpanded to assign icons based on root folder name. This is exactly the API suggested in the discussion above. Closing as completed.
